### PR TITLE
Add tests for era 0 to calendar tests

### PR DESCRIPTION
**Pull request by hughbe**
- Cleans up some namespaces I forgot to clean
- Deletes a file that visual studio never actually deleted from a previous cleanup
- Moves around some of the Calendars.cs and MiscCalendars.cs tests

### CODE REVIEW

File: src/System.Globalization.Calendars/tests/Misc/Calendars.cs [hunk 1]
```diff
@@ -2,27 +2,13 @@
...
         [Fact]
-        public static void GregorianTest()
-        {
-            GregorianCalendar cal = new GregorianCalendar();
-            cal = new GregorianCalendar(GregorianCalendarTypes.Arabic);
-            cal = new GregorianCalendar(GregorianCalendarTypes.Localized);
-            cal = new GregorianCalendar(GregorianCalendarTypes.MiddleEastFrench);
-            cal = new GregorianCalendar(GregorianCalendarTypes.TransliteratedEnglish);
-            cal = new GregorianCalendar(GregorianCalendarTypes.TransliteratedFrench);
-            cal = new GregorianCalendar(GregorianCalendarTypes.USEnglish);
```
Comment (stephentoub) [inline src/System.Globalization.Calendars/tests/Misc/Calendars.cs]: Where did these get moved to?  I don't see them anywhere in the diff.

Comment (hughbe) [inline src/System.Globalization.Calendars/tests/Misc/Calendars.cs]: They already exist in GregorianCalendarCtor.cs

Comment (stephentoub) [inline src/System.Globalization.Calendars/tests/Misc/Calendars.cs]: I see.  Ok.

File: src/System.Globalization.Calendars/tests/Misc/Calendars.cs [hunk 2]
```diff
@@ -60,59 +46,24 @@ public static void JapaneseTest()
...
         {
             KoreanLunisolarCalendar kls = new KoreanLunisolarCalendar();
             Assert.Equal(1, kls.Eras.Length);
         }
 
         [Fact]
-        public static void PersianTest()
-        {
-            PersianCalendar pc = new PersianCalendar();
-            Assert.Equal(1, pc.Eras.Length);
```
Comment (stephentoub) [inline src/System.Globalization.Calendars/tests/Misc/Calendars.cs]: Where is this test duplicating a check for the PersianCalendar eras?

File: src/System.Globalization.Calendars/tests/Misc/MiscCalendars.cs [hunk 3]
```diff
@@ -39,35 +37,11 @@ public static void HijriTest()
...
             DateTime dTest = jCal.ToDateTime(1, 1, 8, 0, 0, 0, 0);
             Assert.Equal(dTest, new DateTime(1989, 1, 8));
         }
-
-        [Fact]
-        public static void KoreanTest()
-        {
-            KoreanCalendar jCal = new KoreanCalendar();
-            DateTime dTest = jCal.ToDateTime(3933, 1, 1, 0, 0, 0, 0);
-            Assert.Equal(dTest, new DateTime(1600, 1, 1));
```
Comment (stephentoub) [inline src/System.Globalization.Calendars/tests/Misc/MiscCalendars.cs]: Can you point me to what this test is duplicating?  I don't see this 3933 number anywhere else in the project, other than in a PersianCalendar test.

File: src/System.Globalization.Calendars/tests/Misc/MiscCalendars.cs [hunk 4]
```diff
@@ -39,35 +37,11 @@ public static void HijriTest()
...
-        }
-
-        [Fact]
-        public static void ThaiTest()
-        {
-            ThaiBuddhistCalendar tCal = new ThaiBuddhistCalendar();
-            DateTime dTest = tCal.ToDateTime(2143, 1, 1, 0, 0, 0, 0);
-            Assert.Equal(dTest, new DateTime(1600, 1, 1));
```
Comment (stephentoub) [inline src/System.Globalization.Calendars/tests/Misc/MiscCalendars.cs]: Ditto, searching for 2143

File: src/System.Globalization.Calendars/tests/Misc/Calendars.cs (hunk 2 above)
Comment (stephentoub) [inline src/System.Globalization.Calendars/tests/Misc/Calendars.cs]: Ah, nevermind, I see you added it in this PR at https://github.com/dotnet/corefx/pull/7743/files#diff-581f48459778b4a7e1b5a20f05e2496aR143

**Comment by stephentoub** [comment]
A couple more questions, otherwise LGTM

File: src/System.Globalization.Calendars/tests/Misc/Calendars.cs (hunk 2 above)
Comment (hughbe) [inline src/System.Globalization.Calendars/tests/Misc/Calendars.cs]: Yup sorry about that, won't happen agaim

File: src/System.Globalization.Calendars/tests/Misc/MiscCalendars.cs (hunk 3 above)
Comment (hughbe) [inline src/System.Globalization.Calendars/tests/Misc/MiscCalendars.cs]: Yup sorry again for not leaving a comment explainimg - the test for ToDateTime for calendars like Korean or ThaiBuddhist or Taiwan (where the year is offset) calculate the year by adding/subtracting the offset front be Gregorian year

See
https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Calendars/tests/KoreanCalendar/KoreanCalendarToDateTime.cs

E.g. 1600+2333=3933

File: src/System.Globalization.Calendars/tests/Misc/MiscCalendars.cs (hunk 4 above)
Comment (hughbe) [inline src/System.Globalization.Calendars/tests/Misc/MiscCalendars.cs]: Ditto with the above:
https://github.com/dotnet/corefx/blob/master/src/System.Globalization.Calendars/tests/ThaiBuddhistCalendar/ThaiBuddhistCalendarToDateTime.cs

1600+543=2143

**Pull request merged**